packages/postgres-database/tests/products/test_utils_products.py

```diff
@@ -97,3 +98,30 @@ async def test_get_or_create_group_product(
                 conn, product_name=product_row.name
             )
             assert product_group_id is None
+
+
+@pytest.mark.testit
+async def test_get_or_create_group_product_concurrent(
+    pg_engine: Engine, make_products_table: Callable
+):
+    async with pg_engine.acquire() as conn:
+        await make_products_table(conn)
+
+    async def _auto_create_products_groups():
+        async with pg_engine.acquire() as conn:
+            async for product_row in await conn.execute(
+                sa.select([products.c.name, products.c.group_id]).order_by(
+                    products.c.priority
+                )
+            ):
+                # get or create
+                product_group_id = await get_or_create_product_group(
+                    conn, product_name=product_row.name
+                )
+                return product_group_id
+
+    tasks = [asyncio.create_task(_auto_create_products_groups()) for _ in range(5)]
```

Comment:
Maybe having this very high, will point out possible issues, and if it works like that it should always be fine.
```suggestion
    tasks = [asyncio.create_task(_auto_create_products_groups()) for _ in range(100)]
```